### ISSUE TITLE
[Chore]: Configure the e2e tests to run via git webhook

### ISSUE THREAD

**Issue by georgethebeatle** [cloudfoundry/korifi#1499]
The github pr concourse resource is a bit lousy at detecting new PRs and especially when there is a large number of PRs it is quite likely that it will miss some and its corresponding check will remain in the Pending state preventing us from merging the PR. The workaround is to rebase the pr or do an empty force push in order to re-trigger all checks.

We are hoping that if we configure the tests to run with a git webhook this problem will go away

**Comment by georgethebeatle** [comment]
Looks like a github webhook will need a server to receive the payload when the desired event occurred. Concourse cannot do this and it is not a planned feature as explained in [this comment](https://github.com/concourse/concourse/issues/4955#issuecomment-570318078). We could write a small app to process the event and trigger the desired build, but we are not convinced it is worth the investment and are concerned that such an app will have to be exposed to the internet and might become a potential attack point. (Alothough there is a way to validate shas sent from Github and detect tampering).

Putting down for now.

**Comment by georgethebeatle** [comment]
I decided to give this chore one more try today, because I found this tool: https://github.com/adnanh/webhook

It is a generic server to handle github webhooks. We could deploy that as a cf app on one of our clusters and it would be available in the internet for Github to POST to. It looks like it handles the security aspect of comparing sha fingerprints. It is even available as a service on www.hookdoo.com.

However the big problem of the push approach as opposed to the concourse default pull approach is that even if we are able to filter the event down to "the label allow-e2es is set on a PR" there is still no way to make concourse run against a sha passed by our webhook server. There is no way to tell concourse "run this job against that commit that I am sending you in the request". Concourse works more like "I will run that job against the latest commit that I know about"

There is also the possibility of setting a [webhook_token](https://concourse-ci.org/resources.html#schema.resource.webhook_token) on the git resource which will allow GitHub to initiate the resource check (instead of the default periodic pull behaviour of concourse), but in this case there is virtually no way to pass parameters down. This means that our best shot is to trigger the run-pr-e2es job every time anyone adds or removes or modifies a label on a PR, issue, etc. And again the job will run against the latest commit in its cache, rather that against the commit associated with the event that triggered it. Needless to say that this is going to generate a lot of load for our concourse instance.

TL;DR It seems to be close to impossible to trigger the PR e2e tests actively from the GitHub side. I am closing that chore.

**Comment by georgethebeatle** [comment]
@kieron-dev feel free to reopen if you have some idea we did not explore.

**Comment by georgethebeatle** [comment]
One possibility might actually be that we deploy `webhook` as an app with an appropriate config that only reacts to newly added labels with the name `allow-e2e` and run the following script:

```
fly -t example pin-resource --resource main/korifi-pr --version ref:bceaf
trap EXIT "fly -t example unpin-resource --resource main/korifi-pr"
fly trigger-job --job main/run-e2es-pr
```

This would cause the correct PR ref to be used, but will also mean that we will have to process github events serially. In any event I think this is too ugly to pursue and we should maybe re-raise our issue with certain PRs getting dropped with the concourse team
